src/Microsoft.DotNet.Build.Tasks/Targets/tests.targets

```diff
@@ -1,32 +1,26 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
-  <UsingTask AssemblyFile="$(PackagesDir)xunit.runners.2.0.0-beta5-build2785\tools\xunit.runner.msbuild.dll" TaskName="Xunit.Runner.MSBuild.xunit" />
-  
-  <ItemGroup>
-    <_SkipTestAssemblies Include="$(SkipTestAssemblies)" />
-  </ItemGroup>
-
-  <Target Name="FindTestDirectories">
+  <Target Name="DiscoverTests">
     <ItemGroup>
-      <AllTestAssemblies Include="$(BaseOutputPathWithConfig)*Tests\*Tests.dll" />
-      <AllTestAssembliesWithParent Include="@(AllTestAssemblies)">
-        <ParentDir>$([System.IO.Path]::GetDirectoryName(%(AllTestAssemblies.Identity)))</ParentDir>
-      </AllTestAssembliesWithParent>
-      <TestDirectories Include="@(AllTestAssembliesWithParent -> '%(ParentDir)')" Exclude="@(_SkipTestAssemblies -> '$(BaseOutputPathWithConfig)%(Identity)')" />
+      <TestAssemblies Include="$(TestWorkingDir)*.Tests\**\*.Tests.dll" Exclude="$(SkipTestAssemblies)" />
+      <TestDirectories Include="@(TestAssemblies -> '%(RootDir)%(Directory)')" KeepDuplicates="false" />
```

Comment:
The directories are the name of the project - let me double check the Exclude logic here works since I removed those skipped tests to port them.